File: atinject-core/src/main/java/org/atinject/core/websocket/dto/WebSocketResponse.java
package org.atinject.core.websocket.dto;

import java.util.UUID;

import org.atinject.core.dto.DTO;

public abstract class WebSocketResponse extends DTO {
	
    private static final long serialVersionUID = 1L;

    private UUID requestId;
    
    private WebSocketResponseFault fault;
    
    public UUID getRequestId() {
        return requestId;
    }

    public WebSocketResponse setRequestId(UUID requestId) {
        this.requestId = requestId;
        return this;
    }

	public WebSocketResponseFault getFault() {
		return fault;
	}

	public void setFault(WebSocketResponseFault fault) {
		this.fault = fault;
	}

}

File: atinject-integration/src/main/resources/webapp/scripts/atinject/WebSocketResponse.js
define(["atinject/DTO", "atinject/WebSocketResponseException"],
function(DTO, WebSocketResponseException){
    function WebSocketResponse() {
        this._class = "org.atinject.core.websocket.dto.WebSocketResponse"
    };

    WebSocketResponse.prototype = new DTO();

        WebSocketResponse.prototype.getRequestId = function() {
        return this.requestId;
    };

    WebSocketResponse.prototype.setRequestId = function(requestId) {
        this.requestId = requestId;
        return this;
    };

    WebSocketResponse.prototype.getException = function() {
        return this.exception;
    };

    WebSocketResponse.prototype.setException = function(exception) {
        this.exception = exception;
        return this;
    };


    
    return WebSocketResponse;
});
